### PR TITLE
seaborn: mark simple deprecations with typing_extensions.deprecated

### PR DESCRIPTION
**Pull request by hamdanal**
Note that I intentionally left out deprecated parameters of the big API functions as they would require overloads with very long signatures.

### CODE REVIEW

File: stubs/seaborn/seaborn/algorithms.pyi [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
     *args: ArrayLike,
     n_boot: int = 10000,
     func: str | Callable[..., Any] = "mean",
     axis: int | None = None,
     units: ArrayLike | None = None,
     seed: _Seed | None = None,
-    random_seed: _Seed | None = None,  # deprecated
+) -> NDArray[Any]: ...
+@overload
+@deprecated("Parameter `random_seed` is deprecated in favor of `seed`")
```
Comment (hamdanal) [inline stubs/seaborn/seaborn/algorithms.pyi:19]: https://github.com/mwaskom/seaborn/blob/2bb945c01ad0c18f4c9faa516f0a75e7b89043ec/seaborn/algorithms.py#L45-L49

File: stubs/seaborn/seaborn/axisgrid.pyi [hunk 2]
```diff
@@ -92,6 +92,7 @@ class _BaseGrid:
         **kwargs: Any,
     ) -> Self: ...
     @property
+    @deprecated("Attribute `fig` is deprecated in favor of `figure`")
```
Comment (hamdanal) [inline stubs/seaborn/seaborn/axisgrid.pyi:95]: https://github.com/mwaskom/seaborn/blob/2bb945c01ad0c18f4c9faa516f0a75e7b89043ec/seaborn/axisgrid.py#L47-L54

File: stubs/seaborn/seaborn/distributions.pyi [hunk 3]
```diff
@@ -139,7 +139,8 @@ def displot(
     facet_kws: dict[str, Any] | None = None,
     **kwargs: Any,
 ) -> FacetGrid: ...
-def distplot(  # deprecated
+@deprecated("Function `distplot` is deprecated and will be removed in seaborn v0.14.0")
```
Comment (hamdanal) [inline stubs/seaborn/seaborn/distributions.pyi:142]: https://github.com/mwaskom/seaborn/blob/2bb945c01ad0c18f4c9faa516f0a75e7b89043ec/seaborn/distributions.py#L2395

File: stubs/seaborn/seaborn/rcmod.pyi [hunk 4]
```diff
@@ -30,10 +30,16 @@ def set_theme(
     color_codes: bool = True,
     rc: dict[str, Any] | None = None,
 ) -> None: ...
-
-# def set(*args, **kwargs) -> None: ...  # deprecated alias for set_theme
-set = set_theme
-
+@deprecated("Function `set` is deprecated in favor of `set_theme`")
```
Comment (hamdanal) [inline stubs/seaborn/seaborn/rcmod.pyi:33]: https://github.com/mwaskom/seaborn/blob/2bb945c01ad0c18f4c9faa516f0a75e7b89043ec/seaborn/rcmod.py#L126-L132

File: stubs/seaborn/seaborn/utils.pyi [hunk 5]
```diff
@@ -75,7 +75,8 @@ def saturate(color: ColorType) -> tuple[float, float, float]: ...
 def set_hls_values(
     color: ColorType, h: float | None = None, l: float | None = None, s: float | None = None
 ) -> tuple[float, float, float]: ...
-def axlabel(xlabel: str, ylabel: str, **kwargs: Any) -> None: ...  # deprecated
+@deprecated("Function `axlabel` is deprecated and will be removed in a future version")
```
Comment (hamdanal) [inline stubs/seaborn/seaborn/utils.pyi:78]: https://github.com/mwaskom/seaborn/blob/2bb945c01ad0c18f4c9faa516f0a75e7b89043ec/seaborn/utils.py#L253-L263

**Comment by hamdanal** [comment]
Is the [stubtest crash](https://github.com/python/typeshed/actions/runs/7152682224/job/19478279459) a known issue?

**Comment by AlexWaygood** [comment]
> Is the [stubtest crash](https://github.com/python/typeshed/actions/runs/7152682224/job/19478279459) a known issue?

Yeah, it's fixed on mypy master, the fix should be included as part of mypy 1.8 :)

**Comment by AlexWaygood** [comment]
See

- https://github.com/python/mypy/pull/16457

**Comment by hamdanal** [comment]
> See
> 
> * [stubtest: special case final and deprecated mypy#16457](https://github.com/python/mypy/pull/16457)

Thank you. I just confirmed that the change passes on mypy's `release-1.8` branch.

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**